### PR TITLE
fixes #661: added option to disable notifications in quiet time

### LINKED ISSUE #661: (Optionally) Disable notifications in Quiet Time
I use quiet time to ignore work e-mails at night. However I can still see the notifications and then I 'must' read the emails. Can this be optionally disabled?

### PR DESCRIPTION
**Pull request by sanderbaas**

### CODE REVIEW

File: k9mail/src/main/java/com/fsck/k9/K9.java [hunk 1]
```diff
@@ -970,6 +973,14 @@ public static void setQuietTimeEnabled(boolean quietTimeEnabled) {
         mQuietTimeEnabled = quietTimeEnabled;
     }
 
+    public static boolean getQuietTimeNotificationEnabled() {
```
Comment (artbristol) [inline k9mail/src/main/java/com/fsck/k9/K9.java]: `isQuietTimeNotificationEnabled` is more idiomatic for small-b `boolean`

File: k9mail/src/main/java/com/fsck/k9/controller/MessagingController.java [hunk 2]
```diff
@@ -4753,6 +4753,11 @@ private LocalMessage findNewestMessageForNotificationLocked(Context context, Not
      */
     private void notifyAccount(Context context, Account account,
             LocalMessage message, int previousUnreadMessageCount) {
+    	// if it's quiet time and notifications are disabled, then we shouldn't show a notification
+        if (K9.isQuietTime() && K9.getQuietTimeNotificationEnabled()) {
```
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/controller/MessagingController.java]: You want to return if it's quiet time and notifications during quiet time are **not** enabled. So this is either a logic error or it's bad naming of the setting/field/method.

**Comment by cketti** [comment]
New global settings also need to be added to `com.fsck.k9.preferences.GlobalSettings` for settings import/export to work.

File: k9mail/src/main/java/com/fsck/k9/controller/MessagingController.java [hunk 3]
```diff
@@ -4753,6 +4753,11 @@ private LocalMessage findNewestMessageForNotificationLocked(Context context, Not
      */
     private void notifyAccount(Context context, Account account,
             LocalMessage message, int previousUnreadMessageCount) {
+    	// if it's quiet time and notifications are disabled, then we shouldn't show a notification
```
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/controller/MessagingController.java]: Please remove the comment. If you feel the condition below isn't clear enough you can do something like the following:
```java
boolean isQuietTimeWithNotificationsDisabled = K9.isQuietTime() && K9.getQuietTimeNotificationEnabled();
if (isQuietTimeWithNotificationsDisabled) {
    return;
}
```

Nobody ever corrects comments when the code changes. By avoiding comments we don't have this particular problem and also the code itself should become more readable.

**Comment by cketti** [comment]
I rebased on master and added the changes myself. See PR #700.

**Comment by sanderbaas** [comment]
Nice! Sorry for being so slow to fix it myself :)

**Pull request closed without merging**